packages/pretty-format/src/index.js

```diff
@@ -316,21 +317,67 @@ function printPlugin(val, indent, prevIndent, spacing, edgeSpacing, refs, maxDep
   return plugin.print(val, boundPrint, boundIndent, opts, colors);
 }
 
-function print(val, indent, prevIndent, spacing, edgeSpacing, refs, maxDepth, currentDepth, plugins, min, callToJSON, printFunctionName, escapeRegex, colors) {
+function print(val: any, indent: string, prevIndent: string, spacing: string, edgeSpacing: string, refs: Refs, maxDepth: number, currentDepth: number, plugins: Plugins, min: boolean, callToJSON: boolean, printFunctionName: boolean, escapeRegex: boolean, colors: Colors): string {
   const basic = printBasicValue(val, printFunctionName, escapeRegex);
-  if (basic) {
+  if (basic !== null) {
     return basic;
   }
 
   const plugin = printPlugin(val, indent, prevIndent, spacing, edgeSpacing, refs, maxDepth, currentDepth, plugins, min, callToJSON, printFunctionName, escapeRegex, colors);
-  if (plugin) {
+  if (plugin !== null) {
     return plugin;
   }
 
   return printComplexValue(val, indent, prevIndent, spacing, edgeSpacing, refs, maxDepth, currentDepth, plugins, min, callToJSON, printFunctionName, escapeRegex, colors);
 }
 
-const DEFAULTS = {
+type Colors = Object;
+type Indent = (str: string) => string;
+type Refs = Array<any>;
+type Serialize = (val: any) => string;
+type StringOrNull = string | null; // but disallow undefined, unlike ?string
+
+type Plugin = {
+  print: (val: any, serialize: Serialize, indent: Indent, opts: Object, colors: Colors) => string,
+  test: Function,
+}
+type Plugins = Array<Plugin>;
+
+type InitialOptions = {|
+  callToJSON?: boolean,
+  escapeRegex?: boolean,
+  highlight?: boolean,
+  indent?: number,
+  maxDepth?: number,
+  min?: boolean,
+  plugins?: Plugins,
+  printFunctionName?: boolean,
+  theme?: {
+    content: string,
+    prop: string,
+    tag: string,
+    value: string,
+  },
+|};
+
+type Options = {|
+  callToJSON: boolean,
+  escapeRegex: boolean,
+  highlight: boolean,
+  indent: number,
+  maxDepth: number,
+  min: boolean,
+  plugins: Plugins,
+  printFunctionName: boolean,
+  theme: {
```

Comment:
Yes, good catch.

In the next pull request, `normalizeOptions` merges `theme` from options argument with default `theme` to fulfill the exact type. Now, an incomplete theme causes plugin to throw.